### DISCUSSION TITLE
modify_body feature explanation

### DISCUSSION THREAD

**Discussion by EastHell** [mitmproxy/mitmproxy#5151] [Q&A] [answered]
Can anyone explain how to use modify_body feature?

I need to modify full body of request with specific url.

1) In config.yaml I printed
```
modify_body: [
  "/~u someUrl.com/some/path & ~q/.*/@~/Documents/requestBody.json"
]
```
2) Then load config file inside mitmproxy
3) Then make new request with url: someUrl.com/some/path 

but url's body doesn't changed. What I'm doing wrong? Ty for any help.

##
```
mitmproxy --version
Mitmproxy: 7.0.4
Python:    3.9.10
OpenSSL:   OpenSSL 1.1.1m  14 Dec 2021
Platform:  macOS-12.1-arm64-arm-64bit
```

**Comment by mhils** [comment, marked as answer]
Have you seen https://docs.mitmproxy.org/dev/overview-features/#modify-body?

One problem I see with your filters is that the separator character (/) is also part of the URL and path. I would recommend picking a different separator. Instead of 
```
/~u someUrl.com/some/path & ~q/.*/@~/Documents/requestBody.json
```
you can for example use
```
#~u someUrl.com/some/path & ~q#.*#@~/Documents/requestBody.json
```

**Reply by EastHell** [reply]
Wow, that's actually working, thank you!
Yes, I'm read this, but I'm not native English speaker and might misunderstood something.

But now instead of replace, I've got double request body (initial one, and right after, one from replacement file)

like:
```
{
    "accountId": "1",
}{
    "accountId": "2",
}
```
instead:
```
{
    "accountId": "2",
}
```

what I'm doing wrong now?

**Reply by EastHell** [reply]
I found the problem. 

The problem was in my body-regex, instead of just` .*`, I should use `.+`.

Thank you so much again @mhils

**Reply by rosydawn6** [reply]
I found this and a few other links looking for information on this topic.  Several attempts were not producing the expected results.  I was using stream_large_bodies=1 and after rereading the docs more https://docs.mitmproxy.org/stable/concepts-options/#stream_large_bodies found that when using stream_large_bodies=1 the body is not stored and hence cannot be modified.
